openraft/src/core/install_snapshot.rs

```diff
@@ -260,13 +260,14 @@ impl<C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> RaftCore<C,
 
         // A local log that is <= last_applied may be inconsistent with the leader.
         // It has to purge all of them to prevent these log form being replicated, when this node becomes leader.
+        self.engine.snapshot_last_log_id = Some(last_applied); // make last applied log removable
         self.engine.purge_log(last_applied);
         self.run_engine_commands::<Entry<C>>(&[]).await?;
 
         self.engine.update_committed_membership(req.meta.last_membership);
         self.run_engine_commands::<Entry<C>>(&[]).await?;
 
-        self.snapshot_last_log_id = self.engine.state.last_applied;
+        self.engine.snapshot_last_log_id = self.engine.state.last_applied;
```

Comment:
`self.engine.snapshot_last_log_id` is assigned twice, is it necessary?

Reply:
if the first assignment is not presented, the engine could not purge to last_applied.(line 264)

Reply:
Then why is the second assignment still needed?

Reply:
Some engine commands were run and I'm afraid these commands will change the value of `self.engine.state.last_applied`.

Reply:
This can be removed. Running engine command should not change `last_applied`.